### PR TITLE
deploy multiple operator fix

### LINKED ISSUE #487: synopsysctl deploy disables CRDs
**What happened**:
 If I do `synopsysctl deploy --enable-alert -n test1` and then `synopsysctl deploy --enable-blackduck -n test2`, the second command will add the blackduck crd but also *remove* the alert crd.

**What you expected to happen**:
Both crds should be in the cluster.

### PR DESCRIPTION
**Pull request by msenmurugan**
* fixes #487 
* fixes for the deployment of opssight CRD without Black Duck CRD

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/511#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_deploy.go [hunk 1]
```diff
@@ -178,6 +184,123 @@ var deployCmd = &cobra.Command{
...
+
+func checkAndUpdateCustomResource(crdType string, namespace string, scope apiextensions.ResourceScope, isClusterScoped bool) (bool, error) {
+	crd, err := util.GetCustomResourceDefinition(apiExtensionClient, crdType)
+	if err != nil {
+		return false, fmt.Errorf("unable to get %s custom resource definition due to %+v", crdType, err)
+	}
+
+	// CRD exist with different scope and hence error out
+	if scope != crd.Spec.Scope {
+		return true, fmt.Errorf("already %s custom resource definition exist with %v scope. updating the %s custom resource definition scope is not supported. please contact the Synopsys support team to handle it", crd.Name, crd.Spec.Scope, crd.Name)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_deploy.go]: "please contact the Synopsys support team to handle it" - leave this off the message.  Could just be a user error

File: pkg/synopsysctl/cmd_deploy.go [hunk 2]
```diff
@@ -178,6 +184,123 @@ var deployCmd = &cobra.Command{
...
+	}
+
+	if isClusterScoped {
+		for key, value := range crd.Labels {
+			if strings.HasPrefix(key, "synopsys.com/operator.") {
+				if value != namespace {
+					return true, fmt.Errorf("already Synopsys Operator is watching for %s custom resource events in namespace '%s'. Only one Synopsys Operator is allowed to watch %s custom resource events. Please contact Synopsys support in case of any queries", crdType, value, crdType)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_deploy.go]: there is already a Synopsys Operator managing {crdType} in namespace {ns}.  Only one Synopsys Operator may manage {crdType} per namespace

^^ Does that sound about right?

File: pkg/synopsysctl/cmd_deploy.go [hunk 3]
```diff
@@ -178,6 +184,123 @@ var deployCmd = &cobra.Command{
...
+				}
+			}
+		}
+	}
+
+	// CRD exist with same scope and hence continue
+	log.Warnf("already %s custom resource definition exist with %v scope and hence adding the namespace operator label to the CRD", crd.Name, crd.Spec.Scope)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_deploy.go]: This isn't a warning so much as informational.  I'd make the log level Info vs Warn.  Also maybe simplify the log message?

"{crdType} with scope {scope} already exists"

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/511#pullrequestreview-252828740" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/511#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_update.go [hunk 4]
```diff
@@ -76,11 +76,10 @@ var updateOperatorCmd = &cobra.Command{
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		// Read Commandline Parameters
+		var err error
 		if len(namespace) > 0 {
-			updateOperator(namespace, cmd)
+			err = updateOperator(namespace, cmd)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_update.go]: Just do "return updateOperator ..." here

File: pkg/synopsysctl/cmd_update.go [hunk 5]
```diff
@@ -76,11 +76,10 @@ var updateOperatorCmd = &cobra.Command{
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		// Read Commandline Parameters
+		var err error
```
Comment (rrati) [inline pkg/synopsysctl/cmd_update.go]: No need to define this

File: pkg/synopsysctl/cmd_update.go [hunk 6]
```diff
@@ -93,21 +92,19 @@ var updateOperatorCmd = &cobra.Command{
 				}
 			}
 			err = updateOperator(operatorNamespace, cmd)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_update.go]: return updateOperator....

File: pkg/synopsysctl/cmd_update.go [hunk 7]
```diff
@@ -93,21 +92,19 @@ var updateOperatorCmd = &cobra.Command{
...
-			if err != nil {
-				return err
-			}
 		}
-		return nil
+		return err
```
Comment (rrati) [inline pkg/synopsysctl/cmd_update.go]: return nil

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/511#pullrequestreview-252828740" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/511#" title="Author self-approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/511#pullrequestreview-252811744" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_deploy.go (hunk 1 above)
Comment (msenmurugan) [reply pkg/synopsysctl/cmd_deploy.go]: done

File: pkg/synopsysctl/cmd_deploy.go (hunk 2 above)
Comment (msenmurugan) [reply pkg/synopsysctl/cmd_deploy.go]: done

File: pkg/synopsysctl/cmd_deploy.go (hunk 3 above)
Comment (msenmurugan) [reply pkg/synopsysctl/cmd_deploy.go]: done

File: pkg/synopsysctl/cmd_update.go (hunk 4 above)
Comment (msenmurugan) [reply pkg/synopsysctl/cmd_update.go]: done

File: pkg/synopsysctl/cmd_update.go (hunk 5 above)
Comment (msenmurugan) [reply pkg/synopsysctl/cmd_update.go]: done

File: pkg/synopsysctl/cmd_update.go (hunk 6 above)
Comment (msenmurugan) [reply pkg/synopsysctl/cmd_update.go]: done

File: pkg/synopsysctl/cmd_update.go (hunk 7 above)
Comment (msenmurugan) [reply pkg/synopsysctl/cmd_update.go]: done

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/511#pullrequestreview-252828740" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/511#" title="Author self-approved">msenmurugan</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/511#pullrequestreview-252859188" title="LGTM">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,msenmurugan,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@rrati: once the present PR merges, I will cherry-pick it on top of release-2019.6.x in a new PR and assign it to you.

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/511#issuecomment-504448691):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by operator-r2d2** [comment]
@rrati: new pull request created: #515

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/511#issuecomment-504448691):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**